chain/client/src/client.rs

```diff
@@ -1966,14 +1671,16 @@ impl Client {
                 .with_label_values(&[&shard_id.to_string()])
                 .start_timer();
             let last_header = Chain::get_prev_chunk_header(epoch_manager, block, shard_id).unwrap();
-            match self.try_produce_chunk(
+            let result = self.chunk_producer.produce_chunk(
```

Comment:
How did we determine if this function `produce_chunks` was supposed to be a part of client.rs or chunk_producer.rs module?

Reply:
Line 1625

Reply:
Again, I didn't want to touch it yet. The main reason is `persist_and_distribute_encoded_chunk` using many client fields